src/scipp/core/cpp_classes.pyi

```diff
@@ -579,7 +579,7 @@ class DataArray:
     def attrs(self) -> Coords:
         ...
 
-    def bin(self, arg_dict: Dict[str, Union[int, Variable]]=None, /, **kwargs: Union[int, Variable]) -> Union[DataArray, Dataset]:
+    def bin(self, arg_dict=None, /, **kwargs):
```

Comment:
Unfortunately we lost type annotations here. But this happens because of the `@overload` declarations, not because of the `data_group_overload` decorator. So it is a general shortcoming of the stubgen that I don't know how to fix in an easy way.